### DISCUSSION TITLE
Backspace is causing a 404

### DISCUSSION THREAD

**Discussion by edalzell** [joshhanley/livewire-autocomplete#42] [Help] [answered]
The implementation:

```html
<x-autocomplete
    class="pt-4 sm:grid sm:grid-cols-3"
    wire:model="movements.{{ $exercise->id }}.id"
    unstyled="true"
>
    <x-autocomplete-label
        class="flex items-center p-0 m-0 mt-2 mb-2 text-sm font-medium leading-5 text-black"
        unstyled="true"
    >
        Exercise {{ $loop->iteration }}:
    </x-autocomplete-label>
    <x-autocomplete-input
        class="block w-full border-0 py-1.5 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-lime-300 sm:text-sm sm:leading-6 font-mono"
        containerClass="relative mt-1 sm:col-span-2"
        wire:model.live="movementSearch.{{ $exercise->id }}"
        unstyled="true"
    />
    <x-autocomplete-list class="mx-2 mt-1" x-cloak>
        <x-autocomplete-new-item wire:click="newMovement('{{ $exercise->id }}')" :value="$movementSearch[$exercise->id]" />
        @foreach ($this->results($exercise->id) as $movement)
            <x-autocomplete-item :key="$movement->id" :value="$movement->name">
                <span>{{ $movement->name }}</span>
            </x-autocomplete-item>
        @endforeach
    </x-autocomplete-list>
</x-autocomplete>
```

When backspace is hit when either the drop down is open OR the field has focus (and the cursor is at the end of the text), I get a 404 from Livewire: 
![CleanShot 2024-07-15 at 16 31 18@2x](https://github.com/user-attachments/assets/f12b4edd-a9e0-4a00-b847-fed391cffaf0)
![CleanShot 2024-07-15 at 16 31 33@2x](https://github.com/user-attachments/assets/ec4361b4-722f-4f19-af5c-5fe246385991)

Any ideas what could be causing it?

**Comment by joshhanley** [comment, marked as answer]
Sorted this on discord.

TL;DR was an updated hook was expecting a selected ID and doing a `findOrFail()`. It didn't take into account being able to clear the selected ID (which backspace does if a record is already selected).
